### DISCUSSION TITLE
bash 4.4  should support  bracketed-paste mode

### DISCUSSION THREAD

**Discussion by longshuaige** [sourcelair/xterm.js#5274] [Q&A]
Starting from version 4.4 of Bash, bracketed paste mode is supported. It seems that even Bash 5 does not support it. iTerm2, however, supports this specification. 

For reference: https://cirw.in/blog/bracketed-paste

**Comment by jerch** [comment]
Posted to wrong repo? Or how is this related to xterm.js? If you think bash lacks in this regard, maybe contact bash maintainers?

xterm.js supports bracketed paste for >5 years. If its not working for, plz file a more concrete issue.

**Comment by longshuaige** [comment]
My point is, with the same version of bash 5.0.17, when I paste multi-line commands, there are brackets in iterm2, but not in xtermjs; if the bash version is a bit higher, such as 5.2, then xtermjs will have bracket decorations.

**Comment by jerch** [comment]
Can you be a bit more specific on what you have tried? A "works there, but not here" isnt very helpful.

**Reply by longshuaige** [reply]
thx

**Comment by Tyriar** [comment]
@longshuaige xterm.js supports it just fine, the problem would be around how you're launching the bash process which is an issue on your end.

**Reply by longshuaige** [reply]
Yes, I noticed that, As long as xtermjs receives the serial number [\ e [? 2004h] sent by bash, it will take effect, but my bash did not send it. thx
